lib/intersection/index.js

```diff
@@ -0,0 +1,29 @@
+import _curry2 from '../_internal/_curry2'
+import equals from '../equals'
+import identical from '../identical'
+
+export default _curry2(function intersection(list1, list2) {
+  var resObj = {}
+
+  for (var i = 0; i < list1.length; i++) {
+    for (var k = 0; k < list2.length; k++) {
```

Comment:
It's not a good idea to use 2 `for` cycles
Need to implement `_internal/_set` and replace `union` as well with it
And also I'd add performance tests like `union` has to check how fast it is.

Reply:
> It's not a good idea to use 2 for cycles

why? the only way to find array's item in another array (except `for` in `for`) is using indexOf - but we need other equality checking that it gives us and also it's the same loop

> And also I'd add performance tests like union has to check how fast it is.

It's a сhallenge, because tests should be complex and comprehensive:
- various length of provided arrays
- various items' types
- various inner objects/arrays length for checking equaling mechanism
- various count of common items
- maybe somethink more

Reply:
I totally agree
But add simple test for input of an array string of different sizes like `union` has to generally understand what time we have for now

Reply:
> why? the only way to find array's item in another array (except for in for) is using indexOf - but we need other equality checking that it gives us and also it's the same loop

That's why I still want to implement `_internal/_set`

Reply:
> That's why I still want to implement _internal/_set

`_set` won't help in removing loops. We need get common elements from both arrays and then we need remove duplicates for it (or all in one iteration)

Reply:
What may be faster than just _dumbest_ `for`?